### PR TITLE
Upgrade `web3-eth` from v1 to v4

### PR DESCRIPTION
**Pull request by bludnic**
- [x] Upgraded `web3-eth` up to v4
- [x] Removed `abi-decoder` dependency since it still uses `web3-eth` v1. Part of the code was moved to the repository, refactored and rewritten in TS.

### CODE REVIEW

**Comment by bludnic** [comment]
[Update Web3 lib for Ethereum](https://trello.com/c/U9RxukKB/367-update-web3-lib-for-ethereum)

**Comment by bludnic** [comment]
@martiliones check now

File: src/lib/abi/abi-decoder.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+type MethodsParams = {
+  name: string
+  type: string
+  value: string | string[]
+}
+
+function componentType(input: Components): string {
+  if (input.type === 'tuple') {
+    const tupleTypes = input.components!.map(componentType)
```
[the remark this answers is not in the document]
Comment (bludnic) [reply src/lib/abi/abi-decoder.ts:19]: Must note that this code is a copy-paste from `abi-decoder` lib which is outdated (it still uses web-eth v1). So better not to touch this code, until we cover it with tests.

File: src/lib/abi/abi-decoder.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    return '(' + tupleTypes.join(',') + ')'
```
[the remark this answers is not in the document]
Comment (bludnic) [reply src/lib/abi/abi-decoder.ts:21]: I think it's already clear what the code does. Don't see a reason to convert into a template literals.

File: src/store/modules/eth/actions.js [hunk 3]
```diff
@@ -75,40 +75,29 @@ const createSpecificActions = (api, queue) => ({
...
-          if (!err) context.commit('blockNumber', number)
-        })
-      ]
-    }
+    // Current gas price
+    void api.getGasPrice().then((price) => {
+      // It is OK with London hardfork
+      context.commit('gasPrice', {
+        gasPrice: price, // string type
+        fee: +(+utils.calculateFee(DEFAULT_ETH_TRANSFER_GAS, price)).toFixed(8) // number type, in ETH
```
[the remark this answers is not in the document]
Comment (bludnic) [reply src/store/modules/eth/actions.js:89]: `utils.calculateFee()` returns string int, so there is no need in converting BigInt -> Number

File: src/lib/abi/abi-decoder.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bludnic) [reply src/lib/abi/abi-decoder.ts:24]: The line 127th looks confused, but using template literals will not make it look better.

**Comment by bludnic** [comment]
Task for refactoring https://trello.com/c/aHNBM6Jq/447-refactor-abidecoder

File: src/store/modules/eth/actions.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bludnic) [reply src/store/modules/eth/actions.js:93]: This line has not changed. The prettier did it.

**Review by RealGoodProgrammer — APPROVED** [review]

**Pull request merged**